### ISSUE TITLE
CA1062 doesn't take NRT into account.

### ISSUE THREAD

**Issue by tacosontitan** [dotnet/roslyn-analyzers#7214]
### Analyzer

**Diagnostic ID**: [CA1062](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1062): `Validate arguments of public methods`

### Analyzer source

Built-in for .NET 8.

### Describe the bug

CA1062 requires validation of arguments for externally exposed methods, so for example:

```cs
protected void Sample(string foo)
{
    // CA1062; validate foo to ensure it isn't null before using it
    foo.ToString();
}
```

However, according to NRT annotations, `foo` isn't `null` here, so why do I need to validate it?

### Steps To Reproduce

1. Enable nullable reference types.
2. Create an externally exposed method with a non-nullable reference type parameter.
3. Attempt to use that parameter without validating it.
4. CA1062 returns positive.

### Expected behavior

CA1062 returns negative due to NRT annotations indicating that the parameter isn't null.

### Actual behavior

CA1062 returned positive.

## Additional context

Related #3740

**Comment by gavinBurtonStoreFeeder** [comment]
You need to validate it because it is still possible to pass null in. Your method is protected which means a derived class, in code you don't control, has access to it.

NRT allows you to flag that certain things are expected to sometimes be null. 

You can't currently block nulls from being passed to any public or private constructor, property or field and thus validation is still required.
